mindmeld/text_preparation/preprocessors.py

```diff
@@ -33,18 +33,37 @@ def process(self, text):
         """
         pass
 
-    @abstractmethod
-    def get_char_index_map(self, raw_text, processed_text):
-        """
-        Generates character index mapping from processed query to raw query.
 
-        See the Tokenizer class for a similar implementation.
+class NoOpPreprocessor(Preprocessor):
+    """
+    NoOpPreprocessor object
+    """
 
+    def process(self, text):
+        """
         Args:
-            raw_text (str)
-            processed_text (str)
+            text (str)
+
+        Returns:
+            (str)
+        """
+        return text
+
 
+class PreprocessorFactory:
+    """Preprocessor Factory Class"""
+
+    @staticmethod
+    def get_preprocessor(preprocessor: str):
+        """A static method to get a Preprocessor
+
+        Args:
+            preprocessor (str): Name of the desired Preprocessor class
         Returns:
-            (dict, dict): A tuple consisting of two maps, forward and backward
+            (Preprocessor): Preprocessor Class
         """
-        pass
+        preprocessor_classes = {NoOpPreprocessor.__name__: NoOpPreprocessor}
+        preprocessor_class = preprocessor_classes.get(preprocessor)
+        if not preprocessor_class:
```

Comment:
If ANLP defines a custom preprocessor, how do we enable that here?

Reply:
To add custom preprocessors without modifying the MindMeld codebase, a developer can make a custom TextPreparationPipeline and pass that in at the top level (Similar to how preprocessors were being passed in before.) I am working on a section in the docs that showcases this functionality.

The other alternative is to modify MindMeld itself and add the classes. I am open to other ideas on this customization as well